fuel-core/src/database.rs

```diff
@@ -248,3 +260,132 @@ impl InterpreterStorage for Database {
         Ok(block.headers.producer)
     }
 }
+
+#[async_trait]
+impl RelayerDb for Database {
+    async fn get_validators(&self) -> HashMap<Address, u64> {
+        struct WrapAddress(pub Address);
+        impl From<Vec<u8>> for WrapAddress {
+            fn from(i: Vec<u8>) -> Self {
+                Self(Address::try_from(i.as_ref()).unwrap())
+            }
+        }
+        let mut out = HashMap::new();
+        for diff in self.iter_all::<WrapAddress, u64>(columns::VALIDATOR_SET, None, None, None) {
+            match diff {
+                Ok((address, stake)) => {
+                    out.insert(address.0, stake);
+                }
+                Err(_) => return out,
+            }
+        }
+        out
+    }
+
+    async fn get_validator_diffs(
+        &self,
+        from_fuel_block: u64,
+        to_fuel_block: Option<u64>,
+    ) -> Vec<(u64, HashMap<Address, u64>)> {
+        let to_fuel_block = if let Some(to_fuel_block) = to_fuel_block {
+            if from_fuel_block > to_fuel_block {
+                return Vec::new();
+            }
+            to_fuel_block
+        } else {
+            u64::MAX
+        };
+        struct WrapU64Be(pub u64);
+        impl From<Vec<u8>> for WrapU64Be {
+            fn from(i: Vec<u8>) -> Self {
+                use byteorder::{BigEndian, ReadBytesExt};
+                use std::io::Cursor;
+                let mut i = Cursor::new(i);
+                Self(i.read_u64::<BigEndian>().unwrap_or_default())
+            }
+        }
+        let mut out = Vec::new();
+        for diff in self.iter_all::<WrapU64Be, HashMap<Address, u64>>(
+            columns::VALIDATOR_SET_DIFFS,
+            None,
+            Some(from_fuel_block.to_be_bytes().to_vec()),
+            None,
+        ) {
+            match diff {
+                Ok((key, diff)) => {
+                    let block = key.0;
+                    if block >= to_fuel_block {
+                        return out;
+                    }
+                    out.push((block, diff))
+                }
+                Err(err) => panic!("get_validator_diffs unexpected error:{:?}", err),
```

Comment:
I'm not really a fan of seeing so many panics, but if it's truly unrecoverable, we should set `panic = "abort"` in our main cargo.toml or configure some kind of panic handler on the service where we'd spawn the relayer task. Otherwise, the relayer task may panic but the rest of the node would remain operational in a silently degraded state. 

https://doc.rust-lang.org/cargo/reference/config.html#profilenamepanic

Reply:
`panic = "abort"` is good idea, in general at least at beggining i realy would like for us to break spectacularly if something goes wrong or unexpected and in the most painful but fastest way to get to the core of the problem. After this first we can relax a litlle bit and have more error frendly way of handing these kind of situations.

Reply:
and in the end i would expect us to have `fuel-core` inside somekind of deamon as if something fails and stops the app it will start again from beggining.